scripts/tools/zap/generate.py

```diff
@@ -106,8 +106,13 @@ def detectZclFile(zapFile):
 
 
 def runArgumentsParser() -> CmdLineArgs:
-    default_templates = 'src/app/zap-templates/app-templates.json'
-    default_output_dir = 'zap-generated/'
+    # By default generate the idl file only. This will get moved from the
+    # output directory into the zap file directory automatically.
+    #
+    # All the rest of the files (app-templates.json) are generally built at
+    # compile time.
+    default_templates = 'src/app/zap-templates/matter-idl.json'
```

Comment:
Can we fix `zap_regen_all.py` to not pass in the now-superfluous `-t` argument when it passes this value?  That would let people continue to use `--dry-run` on that file to discover "sane" commands to run to generate things....

Reply:
Updated, had to alter a bit output directory logic.

Testing a bit more now since it looks to me like zap_regen_all does not fully wait now for zap-cli to finish.

Reply:
Chef seems to require an output directory, since it creates an empty filr for af-gen-event. Need to figure that one out...

Reply:
Updated for chef logic as well. It is 100% hackish, however hoping chef will soon get rid of af-gen-event.